unitTests/ui/FacetTest.ts

```diff
@@ -759,37 +759,143 @@ export function FacetTest() {
+
+        function getMasterAndDependentFacetResults() {
+          const results = FakeResults.createFakeResults();
+          results.groupByResults = [
+            FakeResults.createFakeGroupByResult(dependentFacetField, 'foo', 15),
+            FakeResults.createFakeGroupByResult(masterFacetField, 'foo', 15)
+          ];
+
+          return results;
+        }
+
+        function getMasterDependentFacetStateAttributes() {
+          return {
+            ...getMasterFacetStateAttributes(),
+            ...getDependentFacetStateAttributes()
+          };
+        }
+
+        function getMasterFacetStateAttributes() {
+          return {
+            [`f:${masterFacetField}`]: masterFacet.cmp.getSelectedValues(),
+            [`f:${masterFacetField}:not`]: masterFacet.cmp.getExcludedValues()
+          };
+        }
+
+        function getDependentFacetStateAttributes() {
+          return {
+            [`f:${dependentFacetField}`]: test.cmp.getSelectedValues(),
+            [`f:${dependentFacetField}:not`]: test.cmp.getExcludedValues()
+          };
+        }
+
+        beforeEach(() => {
+          masterFacet = Mock.advancedComponentSetup<Facet>(
+            Facet,
+            new Mock.AdvancedComponentSetupOptions(
+              undefined,
+              {
+                field: masterFacetField
+              },
+              (builder: Mock.MockEnvironmentBuilder) => {
+                return builder.withRoot(test.env.root);
+              }
+            )
+          );
+
+          test = Mock.optionsComponentSetup<Facet, IFacetOptions>(Facet, {
+            field: dependentFacetField,
+            dependsOn: masterFacetField
+          });
+
+          test.cmp.queryStateModel.get = key => {
+            const attributes = getMasterDependentFacetStateAttributes();
+            return attributes[key];
+          };
+
+          Simulate.query(test.env, { results: getMasterAndDependentFacetResults() });
         });
 
-        var masterFacet = Mock.advancedComponentSetup<Facet>(
-          Facet,
-          new Mock.AdvancedComponentSetupOptions(
-            undefined,
-            {
-              field: '@masterFacet'
-            },
-            (builder: Mock.MockEnvironmentBuilder) => {
-              return builder.withRoot(test.env.root);
-            }
-          )
-        );
+        it('adds the coveo-facet-dependent class to the dependent facet', () => {
+          expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(true);
+        });
 
-        var results = FakeResults.createFakeResults();
-        results.groupByResults = [
-          FakeResults.createFakeGroupByResult('@field', 'foo', 15),
-          FakeResults.createFakeGroupByResult('@masterFacet', 'foo', 15)
-        ];
+        it('does not add the coveo-facet-dependent class to the master facet', () => {
+          expect($$(masterFacet.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
```

Comment:
I know this is the original name, but maybe this would be a good opportunity to rename this css class? Maybe to something more specific like coveo-facet-dependent-hidden?

Reply:
True, the name could have been better, but since it's a existing css class (and not a new one we're adding), it's safer if we dont change it unless absolutely needed, for backward compatibility reasons.

Reply:
I will keep the name as-is then.